immich: 1.129.0 -> 1.130.3

Scrumplex opened the pull request:
https://github.com/immich-app/immich/releases/tag/v1.130.0
https://github.com/immich-app/immich/releases/tag/v1.130.1
https://github.com/immich-app/immich/releases/tag/v1.130.2
https://github.com/immich-app/immich/releases/tag/v1.130.3

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

dotlambda reviewed (commented):
> `ln: failed to create symbolic link '/build/bin/c++filt' -> '': No such file or directory`

Remove the appropriate line from `package.nix`.

Scrumplex commented:
Seems like build succeeds without any of those symlinks. Interesting

Scrumplex commented:
immich-machine-learning now requires hatchling during build

rhelmot commented on pkgs/by-name/im/immich/package.nix:
```diff
@@ -166,13 +166,6 @@ buildNpmPackage' {
-    ln -s "$(type -p ${stdenv.cc.targetPrefix}pkg-config)" $TMP/bin/pkg-config
-    ln -s "$(type -p ${stdenv.cc.targetPrefix}c++filt)" $TMP/bin/c++filt
-    ln -s "$(type -p ${stdenv.cc.targetPrefix}readelf)" $TMP/bin/readelf
-    export PATH="$TMP/bin:$PATH"
```
These lines were for cross. I'll have to test if they're still required.

Scrumplex replied:
I was able to build `pkgsCross.aarch64-multiplatform.immich.web` on my `x86_64-linux`

dotlambda commented on pkgs/by-name/im/immich/package.nix:
```diff
@@ -166,13 +166,6 @@ buildNpmPackage' {
-
-    # some part of the build wants to use un-prefixed binaries. let them.
-    mkdir -p $TMP/bin
-    ln -s "$(type -p ${stdenv.cc.targetPrefix}pkg-config)" $TMP/bin/pkg-config
```
Why was this line removed? The build succeeds with it

dotlambda replied:
You should only have removed the two lines which caused the build to fail

Scrumplex replied:
According to @rhelmot's review above, the code I removed was fixing a build failure when cross compiling. But as cross builds still work, I don't see a reason to keep the code around. I could split it up into two commits, in case we decide to revert it again in the future

dotlambda replied:
I see, thanks for the clarification.

tebriel commented:
1.130.1 is now out in case you didn't see!

Titaniumtown reviewed (approved):
diff LGTM besides the cross compiling stuff, which I do not know about.

tebriel commented:
quick check confirms the deploy generally works, haven't tested in depth.

rhelmot committed:
immich: fix npmDeps build failure

It seems like c++filt and readelf are not always in PATH. npmDeps fails to
build because of that.
We add `|| true` to be resilient to this and move to prePatch because
npmConfigHook is a postPatch hook.

Scrumplex committed:
immich: 1.129.0 -> 1.130.3

https://github.com/immich-app/immich/releases/tag/v1.130.0
https://github.com/immich-app/immich/releases/tag/v1.130.1
https://github.com/immich-app/immich/releases/tag/v1.130.2
https://github.com/immich-app/immich/releases/tag/v1.130.3

immich-machine-learning switched from poetry to hatchling

Scrumplex commented:
I have split up my commits in order to document the changes better. 

1. The `c++filt` removal actually fixes a build failure that is present on master, but wasn't hit by hydra, as it's in the already cached FOD.
2. The bump itself. To avoid build failures during bisections, I obviously included the build-system change in the version bump
3. I removed the cross build fix that seems to be unnecessary now

tebriel commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 393108`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>immich</li>
    <li>immich-cli</li>
    <li>immich-machine-learning</li>
    <li>immich-machine-learning.dist</li>
  </ul>
</details>

dotlambda commented:
> immich: fix npmDeps build failure

I don't understand. The build succeeds on master.

In contrast, the build fails at 17dfcf871053321d0faceb2c6232b27b83dc8b80.
I think one commit makes more sense.

Scrumplex commented:
> > immich: fix npmDeps build failure
> 
> I don't understand. The build succeeds on master.

I have checked out the merge base of my branch (9e492172209fb7335438a9b000aa54cfe4efa6fa)

```console
$ git checkout 9e492172209fb7335438a9b000aa54cfe4efa6fa
$ nix build -vLf . immich.npmDeps
$ # ^ cached
$ nix build -vLf . immich.npmDeps --rebuild
checking outputs of '/nix/store/3kh37z9ccbr7r1hbd6lrfb2z9a61vv3z-immich-1.129.0-npm-deps.drv'...
immich> Running phase: unpackPhase
immich> unpacking source archive /nix/store/39blq2vdh77a95b1fmpwjmsisr9hxpri-source/server
immich> source root is server
immich> Running phase: patchPhase
immich> ln: failed to create symbolic link '/build/bin/c++filt' -> '': No such file or directory
error: builder for '/nix/store/3kh37z9ccbr7r1hbd6lrfb2z9a61vv3z-immich-1.129.0-npm-deps.drv' failed with exit code 1;
       last 5 log lines:
       > Running phase: unpackPhase
       > unpacking source archive /nix/store/39blq2vdh77a95b1fmpwjmsisr9hxpri-source/server
       > source root is server
       > Running phase: patchPhase
       > ln: failed to create symbolic link '/build/bin/c++filt' -> '': No such file or directory
       For full logs, run 'nix log /nix/store/3kh37z9ccbr7r1hbd6lrfb2z9a61vv3z-immich-1.129.0-npm-deps.drv'.
```

dotlambda commented:
Oh you're right. It was cached. Anyway, after your first commit the build still fails.

Scrumplex commented:
> Anyway, after your first commit the build still fails.

I see. I also removed the patchelf line in the first commit instead of the last commit now.

rhelmot commented:
My CPU is tied up right now but I will try to rebase this onto my dev branch and do a cross build tonight.

tebriel commented:
my passthru tests hang every time I try to run them:

```
$ nix build --file /nix/store/wyrz9irhdpcqn9lnalzdyx10blf8xs35-nixpkgs-review-3.1.0/lib/python3.12/site-packages/nixpkgs_review/nix/review-shell.nix --nix-path 'nixpkgs=/home/tebriel/.cache/nixpkgs-review/pr-393108-3/nixpkgs nixpkgs-overlays=/tmp/tmpstsp5pqk' --extra-experimental-features 'nix-command no-url-literals' --no-link --keep-going --no-allow-import-from-derivation --option build-use-sandbox relaxed --argstr local-system x86_64-linux --argstr nixpkgs-path /home/tebriel/.cache/nixpkgs-review/pr-393108-3/nixpkgs --argstr nixpkgs-config-path /tmp/tmp9hjmc5sl.nix --argstr attrs-path /home/tebriel/.cache/nixpkgs-review/pr-393108-3/attrs.nix
[1/0/13 built] building python3.12-albumentations-2.0.5 (pytestCheckPhase): tests/test_transforms.py ............................................... [
```

tebriel commented:
busy week for immich
https://github.com/immich-app/immich/releases/tag/v1.130.3

```diff
diff --git a/pkgs/by-name/im/immich/sources.json b/pkgs/by-name/im/immich/sources.json
index 552b7aacd39f..e84018224f45 100644
--- a/pkgs/by-name/im/immich/sources.json
+++ b/pkgs/by-name/im/immich/sources.json
@@ -1,22 +1,22 @@
 {
-  "version": "1.130.2",
-  "hash": "sha256-SVSoQM0xGw2KSoxRq+dsLq5wUGrreIfAhCXxvPZoH6Y=",
+  "version": "1.130.3",
+  "hash": "sha256-qMRX8gwoagZJwhMg/1X+ZItFyC73qxLR8pHujWu1rog=",
   "components": {
     "cli": {
-      "npmDepsHash": "sha256-Kieg2qU0LqpGq5KDuENBh8kFcLS5+fxD0yzgoBonOzo=",
-      "version": "2.2.56"
+      "npmDepsHash": "sha256-l8Aw9V6bwj/kauBmmKGiqytzjGwyXL1qlEg54eQcddQ=",
+      "version": "2.2.57"
     },
     "server": {
-      "npmDepsHash": "sha256-Vyb0zXikKZ41X8brDooQzwhkqzz4MF3GKUrDi825G74=",
-      "version": "1.130.2"
+      "npmDepsHash": "sha256-4FiD1+lVqth3MNzityx3oAo/LLXkfTaA57vG4KUhUcM=",
+      "version": "1.130.3"
     },
     "web": {
-      "npmDepsHash": "sha256-91XGjsGUfCouZgXIB8BX/6CST6NlDp6WAPLXsEbd1r0=",
-      "version": "1.130.2"
+      "npmDepsHash": "sha256-nUMXNdEK8Dyai0/NKcXD8bB+aH/ZeiDcQmRKVzo9cqc=",
+      "version": "1.130.3"
     },
     "open-api/typescript-sdk": {
-      "npmDepsHash": "sha256-FSxhTX5Fc2Ez6WX3jmnmvXV4CuRQrl94UwhCY/JBvhE=",
-      "version": "1.130.2"
+      "npmDepsHash": "sha256-B8NqGLgmT126Cf0uPx4Eka60i3mNPskvHnfVTEqf0BY=",
+      "version": "1.130.3"
     },
     "geonames": {
       "timestamp": "20250325151913",
```

rhelmot commented:
Building now. See you in 75 derivations...

rhelmot commented:
`/nix/store/11ciq72n4fdv8rw6wgjgasfv4mjs1jrw-bash-5.2p37/bin/sh: line 1: pkg-config: command not found`

Seems like the missing lines were necessary after all...

rhelmot commented:
I'm going to find the happy medium between with/without the symlinks and push to this branch when I have it building for both linux and freebsd.

rhelmot commented:
I pushed two commits which allowed me to build for linux and freebsd (cherry picked onto my dev branch which has all the dependencies fixed for freebsd cross). I did not encounter any of the file-not-found errors. I did, however, encounter a hash mismatch. Feel free to remix or squash those commits in any way, I don't care about my commit authorship.

dotlambda commented:
> I did not encounter any of the file-not-found errors.

I still get
```
ln: failed to create symbolic link '/build/bin/c++filt' -> '': No such file or directory
```
when I run `nix-build -A immich.npmDeps`.
This might be related to `fetchNpmDeps` using `stdenvNoCC`.
Can you try replacing `postPatch` with `preBuild`?

rhelmot commented:
Extremely concerning. Can you post the derivation hash and git hash you have?

dotlambda commented:
> Extremely concerning. Can you post the derivation hash and git hash you have?

I'm pretty sure you can reproduce it by running `nix-build -A immich.npmDeps --check`.
On bdb4f653e640 I get
```
$ nix-build -A immich.npmDeps
this derivation will be built:
  /nix/store/qa60w5n5qnqsh40haj88lxjh02njsc1j-immich-1.130.2-npm-deps.drv
building '/nix/store/qa60w5n5qnqsh40haj88lxjh02njsc1j-immich-1.130.2-npm-deps.drv' on 'ssh://nix@willard'...
copying 0 paths...
Running phase: unpackPhase
unpacking source archive /nix/store/84n3v1gmip4b0m09l25sdilwc8h9qrs4-source/server
source root is server
Running phase: patchPhase
ln: failed to create symbolic link '/build/bin/c++filt' -> '': No such file or directory
error: build of '/nix/store/qa60w5n5qnqsh40haj88lxjh02njsc1j-immich-1.130.2-npm-deps.drv' on 'ssh://nix@willard' failed: builder for '/nix/store/qa60w5n5qnqsh40haj88lxjh02njsc1j-immich-1.130.2-npm-deps.drv' failed with exit code 1;
       last 5 log lines:
       > Running phase: unpackPhase
       > unpacking source archive /nix/store/84n3v1gmip4b0m09l25sdilwc8h9qrs4-source/server
       > source root is server
       > Running phase: patchPhase
       > ln: failed to create symbolic link '/build/bin/c++filt' -> '': No such file or directory
       For full logs, run 'nix-store -l /nix/store/qa60w5n5qnqsh40haj88lxjh02njsc1j-immich-1.130.2-npm-deps.drv'.
error: builder for '/nix/store/qa60w5n5qnqsh40haj88lxjh02njsc1j-immich-1.130.2-npm-deps.drv' failed with exit code 1;
       last 5 log lines:
       > Running phase: unpackPhase
       > unpacking source archive /nix/store/84n3v1gmip4b0m09l25sdilwc8h9qrs4-source/server
       > source root is server
       > Running phase: patchPhase
       > ln: failed to create symbolic link '/build/bin/c++filt' -> '': No such file or directory
       For full logs, run 'nix-store -l /nix/store/qa60w5n5qnqsh40haj88lxjh02njsc1j-immich-1.130.2-npm-deps.drv'.
```

rhelmot commented:
Oh, the FOD... yeah, I can reproduce that. One sec.

rhelmot commented:
So the problematic code (npmConfigHook) is called as a postPatch hook, which means that we actually have to do this as probably prePatch if we want to be sure about this! I'm going to just slap some || true clauses on it and call it a day since if the executables aren't in PATH we're going to fail anyway :P

dotlambda commented:
I squashed the commits as I saw fit and updated immich to 1.130.3.

tebriel commented:
> ```
> error: build of '/nix/store/q6hvl35acfkkyv9krgg49i3ih1zpy5mc-immich-geodata.drv' on 'ssh-ng://bob-the-builder@eclipse.sefa.cloud' failed: hash mismatch in fixed-output derivation '/nix/store/q6hvl35acfkkyv9krgg49i3ih1zpy5mc-immich-geodata.drv':
>          specified: sha256-u6m6gSFBMtVmknS/adj1R8GcU8IHtebdDWBkrVeIXio=
>             got:    sha256-bg+KON+ydWlRJ8MsIL7Ue0c97aqiiItuK8+kFngd6W0=
> error: builder for '/nix/store/q6hvl35acfkkyv9krgg49i3ih1zpy5mc-immich-geodata.drv' failed with exit code 1
> ```
> 
> I'd hope that Archive.org is reproducible :/

it's not, this happens all the time, usually takes a few days to a week for the sha to stabilize in my upgrade experience for immich I haven't looked into why it keeps changing (guessing we just keep getting latest until it eventually ends up in an archive?)

vale981 commented:
> I squashed the commits as I saw fit and updated immich to 1.130.3.

(update title of pr?)

thanks for your work!
